lib/ui/pages/score/widgets/score_tile_widget.dart

```diff
@@ -0,0 +1,83 @@
+// ignore_for_file: import_of_legacy_library_into_null_safe
+
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_app/src/model/course/course_score_json.dart';
+
+typedef OnCategoryChanged = void Function(int? category);
+
+class ScoreTile extends StatelessWidget {
+  ScoreTile({
+    super.key,
+    required String courseName,
+    required String category,
+    required String scoreValue,
+    OnCategoryChanged? onCategoryChanged,
+  })  : _courseName = courseName,
+        _category = category,
+        _scoreValue = scoreValue,
+        _onCategoryChanged = onCategoryChanged;
+
+  /// The score value of a course.
+  /// Note that we should make the score's type to be a [String] instead of [int] since the score can be a string like "Q".
+  final String _scoreValue;
+  final String _category;
+  final String _courseName;
+  final OnCategoryChanged? _onCategoryChanged;
+
+  final ValueNotifier<int?> _selectedCategory = ValueNotifier(null);
+
+  int? _getInitialCategoryIndex() {
+    final index = constCourseType.indexOf(_category);
+    return index == -1 ? null : index;
+  }
+
+  Widget _buildCourseName() => AutoSizeText(
+        _courseName,
+        style: const TextStyle(fontSize: 16.0),
+      );
+
+  Widget _buildCategoryMenu() => ValueListenableBuilder(
+        valueListenable: _selectedCategory,
+        builder: (_, value, __) => DropdownButton(
+          underline: const SizedBox.shrink(),
+          value: value ?? _getInitialCategoryIndex(),
+          items: constCourseType
+              .asMap()
+              .entries
+              .map((category) => _buildCategoryMenuItem(category.value, category.key))
+              .toList(),
+          onChanged: (newCategory) {
+            _selectedCategory.value = newCategory;
+            _onCategoryChanged?.call(newCategory);
+          },
+        ),
+      );
+
+  DropdownMenuItem<int> _buildCategoryMenuItem(String category, int index) => DropdownMenuItem(
+        value: index,
+        child: Text(
+          category,
+          style: const TextStyle(fontSize: 16.0),
+        ),
+      );
+
+  Widget _buildScoreValue() => SizedBox(
```

Comment:
Suggest changing to getter and rename variable.

```suggestion
  Widget get _scoreValueText => SizedBox(
```

Reply:
addressed in [633d105](https://github.com/NEO-TAT/tat_flutter/pull/157/commits/633d1059b553de0dbb6a9d28b7a255fe3db42e56).